### PR TITLE
Adding blender and globalprotect with nixGL prefixing

### PR DESCRIPTION
**Pull request by jlilja**

### CODE REVIEW

**Comment by jlilja** [comment]
@Lillecarl what do you think about this approach to support applications that requires openGL support? For HM it is not provided out of the box so I needed to fix it somehow.

**Comment by Lillecarl** [comment]
I'll hake a look, I'm out and about this weekend. Monday! 👌

File: jonas/desktop/custom.nix [hunk 1]
```diff
@@ -0,0 +1,14 @@
+{ config, pkgs, ... }:
+
```
Comment (Lillecarl) [inline jonas/desktop/custom.nix:2]:
```
let
  nixgl = "${nixgl.nixGLIntel}/bin/nixGLIntel";
in
```

There are probably other ways to wrap programs, but that one gives you ultimate flexibility

File: jonas/desktop/custom.nix [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+{
+  home.packages = with pkgs; [
+    (writeShellScriptBin "blender" ''
+      ${nixgl.nixGLIntel}/bin/nixGLIntel ${blender}/bin/blender
```
Comment (Lillecarl) [inline jonas/desktop/custom.nix:6]: add $@ to the end, that passes all CLI arguments to blender

File: jonas/desktop/custom.nix [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+    '')
+
+
+    (writeShellScriptBin "globalprotect" ''
+      ${nixgl.nixGLIntel}/bin/nixGLIntel ${globalprotect-openconnect}/bin/gpclient
```
Comment (Lillecarl) [inline jonas/desktop/custom.nix:11]: add $@ to the end, that passes all CLI arguments to gpclient (resign from Viaplay already)

File: readme.md [hunk 4]
```diff
@@ -19,14 +19,6 @@ Nix configuration with home manager.
 6. Clone this repo in `$HOME/.config/home-manager`.
 7. `home-manager switch --flake .`
 
-## Commands
-
-`home-manager switch --flake .` - Evaluates the home manager configuration and provisions the specified packages in flake.nix.
-
-`nix-collect-garbage -d` - Clears up previous generations generated by home-manager switch.
```
Comment (Lillecarl) [inline readme.md:26]: Remember to expire home-manager generations too (see https://github.com/Lillecarl/nixos/blob/master/clean)

File: flake.nix [hunk 5]
```diff
@@ -11,6 +11,8 @@
       inputs.nixpkgs.follows = "nixpkgs";
     };
 
+    nixGL.url = "github:guibou/nixGL";
```
Comment (Lillecarl) [inline flake.nix:14]:
```
nixGL = {
  url = "github:guibou/nixGL";
  inputs.nixpkgs.follows = "nixpkgs";
};
```
You don't wanna copy nixpkgs too much

**Review by Lillecarl — COMMENTED** [review]
Comments, overall looks nice!

Did you know you can install casks with HM? 😄

File: flake.nix (hunk 5 above)
Comment (jlilja) [reply flake.nix:14]: Will it create a copy of nixpkgs if I'm not explicit in what it should follow?

File: jonas/desktop/custom.nix (hunk 3 above)
Comment (jlilja) [reply jonas/desktop/custom.nix:11]: lol

File: jonas/desktop/custom.nix (hunk 1 above)
Comment (jlilja) [reply jonas/desktop/custom.nix:2]: Ah, makes sense

File: jonas/desktop/custom.nix (hunk 2 above)
Comment (jlilja) [reply jonas/desktop/custom.nix:6]: Great! Will do

File: flake.nix (hunk 5 above)
Comment (Lillecarl) [reply flake.nix:14]: Yup, any dependencies of nixgl would be duplicated all the way down to stdenv (doesn't have to be a lot, but could be multiple glibc, python, bash yadayada

**Pull request merged**